### PR TITLE
Improve implementation of parser for tuple project

### PR DESCRIPTION
**Pull request by ajreynol**
This PR refactors how special cases of indexed operators are parsed.

### CODE REVIEW

**Review by mpreiner — APPROVED** [review]

File: src/parser/smt2/Smt2.g [hunk 1]
```diff
@@ -1703,7 +1697,36 @@ identifier[cvc5::ParseOp& p]
...
     | sym=SIMPLE_SYMBOL nonemptyNumeralList[numerals]
       {
-        p.d_op = PARSER_STATE->mkIndexedOp(AntlrInput::tokenText($sym), numerals);
+        std::string opName = AntlrInput::tokenText($sym);
+        api::Kind k = PARSER_STATE->getIndexedOpKind(opName);
+        if (k==api::APPLY_SELECTOR || k==api::APPLY_UPDATER)
+        {
+          // we adopt a special syntax (_ tuple_select n) and (_ tuple_update n)
+          // for tuple selectors and updaters
+          if (numerals.size()!=1)
```
Comment (mpreiner) [inline src/parser/smt2/Smt2.g:1706]:
```suggestion
          if (numerals.size() != 1)
```

File: src/parser/smt2/Smt2.g [hunk 2]
```diff
@@ -1703,7 +1697,36 @@ identifier[cvc5::ParseOp& p]
       }
     | sym=SIMPLE_SYMBOL nonemptyNumeralList[numerals]
       {
-        p.d_op = PARSER_STATE->mkIndexedOp(AntlrInput::tokenText($sym), numerals);
+        std::string opName = AntlrInput::tokenText($sym);
+        api::Kind k = PARSER_STATE->getIndexedOpKind(opName);
+        if (k==api::APPLY_SELECTOR || k==api::APPLY_UPDATER)
```
Comment (mpreiner) [inline src/parser/smt2/Smt2.g:1702]:
```suggestion
        if (k == api::APPLY_SELECTOR || k == api::APPLY_UPDATER)
```

**Review by mpreiner — APPROVED** [review]
LGTM, just two minor code formatting suggestions.

File: src/parser/smt2/Smt2.g (hunk 2 above)
Comment (mudathirmahgoub) [reply src/parser/smt2/Smt2.g]: Please add an example for `tuple_update` to one of the regressions like other tuple operators.

**Comment by ajreynol** [comment]
This is low priority, and will take some effort to debug, closing for now.

**Pull request closed without merging**